packages/app/src/analytic-data-tracker/services/tracker-cache.service.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import path from "path"
+import { FileSystemService } from "App/file-system/services/file-system.service.refactored"
+import getAppPath from "App/main/utils/get-app-path"
+
+export class TrackerCacheService {
+  private readonly filePath = path.join(getAppPath(), "track-cache.txt")
+
+  constructor(private fileSystem: FileSystemService) {
+    void this.init()
+  }
+
+  public async isEventUnique(event: Object): Promise<boolean> {
+    const cache = await this.fileSystem.readFile(this.filePath)
+    return !cache.toString().includes(JSON.stringify(event))
+  }
+
+  public async saveEvent(event: Object): Promise<void> {
+    await this.fileSystem.appendFile(this.filePath, `${JSON.stringify(event)};`)
```

Comment:
Did you think about the situation when this file became larger and longer, without any `rotation` mechanism, it can memory leaks

Reply:
https://github.com/mudita/mudita-center/pull/693/commits/841e39c00790841cdfb2f8d4fc5a3e21ad272d6b